aptos-move/framework/aptos-experimental/sources/trading/market/market.move

```diff
@@ -555,6 +516,140 @@ module aptos_experimental::market {
+                callbacks
+            );
+            return option::none();
+        };
+        let fill_id = self.next_fill_id();
+        let settle_result =
+            callbacks.settle_trade(
+                user_addr,
+                maker_address,
+                order_id,
+                maker_order_id,
+                fill_id,
+                is_bid,
+                maker_order.get_price(), // Order is always matched at the price of the maker
+                maker_matched_size,
+                metadata,
+                maker_order.get_metadata_from_order()
+            );
+
+        let unsettled_maker_size = maker_matched_size;
+        let settled_size = settle_result.get_settled_size();
+        if (settled_size > 0) {
+            *remaining_size -= settled_size;
+            unsettled_maker_size -= settled_size;
+            fill_sizes.push_back(settled_size);
+            // Event for taker fill
+            self.emit_event_for_order(
+                order_id,
+                user_addr,
+                orig_size,
+                *remaining_size,
+                settled_size,
+                maker_order.get_price(),
+                is_bid,
+                true,
+                ORDER_STATUS_FILLED,
+                &std::string::utf8(b"")
+            );
+            // Event for maker fill
+            self.emit_event_for_order(
+                maker_order_id,
+                maker_address,
+                maker_order.get_orig_size(),
+                maker_order.get_remaining_size() + unsettled_maker_size,
+                settled_size,
+                maker_order.get_price(),
+                !is_bid,
+                false,
+                ORDER_STATUS_FILLED,
+                &std::string::utf8(b"")
+            );
+        };
+
+        let maker_cancellation_reason = settle_result.get_maker_cancellation_reason();
+        if (maker_cancellation_reason.is_some()) {
+            self.cancel_maker_order_internal(
+                &maker_order,
+                maker_order_id,
+                maker_address,
+                maker_cancellation_reason.destroy_some(),
+                unsettled_maker_size,
+                callbacks
+            );
+        };
+
+        let taker_cancellation_reason = settle_result.get_taker_cancellation_reason();
+        if (taker_cancellation_reason.is_some()) {
+                self.cancel_order_internal(
+                    user_addr,
+                    price,
+                    order_id,
+                    orig_size,
+                    *remaining_size,
+                    *fill_sizes,
+                    is_bid,
+                    true, // is_taker
+                    OrderCancellationReason::ClearinghouseSettleViolation,
+                    taker_cancellation_reason.destroy_some(),
+                    callbacks
+                );
+            if (maker_cancellation_reason.is_none() && unsettled_maker_size > 0) {
```

Comment:
You don't need to check maker cancellation reason here cuz it's already checked in line 597?